test/functional/shared.js

```diff
@@ -276,6 +276,23 @@ class APMEventCollector {
   }
 }
 
+// simplified `withClient` helper that only uses callbacks
+function withClientV2(callback) {
```

Comment:
This will be great to use going forward, why not add a `@deprecation` warning to the old `withClient`, it'll get the ~strikethrough~ in VSCode to help remind the next dev that this exists 😃

Reply:
tbh, I'm not sure why the existing `withClient` didn't work, so there's likely more work that could go into merging them into a common method. This method doesn't support promises at all, for instance, so that's probably undesirable :man_shrugging:

Reply:
Ah fair, fair 👍